k9mail/src/main/java/com/fsck/k9/ui/ContactBadge.java

```diff
@@ -0,0 +1,303 @@
+package com.fsck.k9.ui;
+
+import android.content.Context;
+import android.os.Bundle;
+import android.util.AttributeSet;
+import android.view.accessibility.AccessibilityEvent;
+import android.view.accessibility.AccessibilityNodeInfo;
+import android.widget.ImageView;
+
+import android.content.AsyncQueryHandler;
+import android.content.ContentResolver;
+import android.content.Intent;
+import android.database.Cursor;
+import android.graphics.Canvas;
+import android.graphics.drawable.Drawable;
+import android.net.Uri;
+import android.provider.ContactsContract.CommonDataKinds.Email;
+import android.provider.ContactsContract.Contacts;
+import android.provider.ContactsContract.Intents;
+import android.provider.ContactsContract.PhoneLookup;
+import android.provider.ContactsContract.QuickContact;
+import android.provider.ContactsContract.RawContacts;
+import android.view.View;
+import android.view.View.OnClickListener;
+
+/**
+ * ContactBadge replaces the android ContactBadge for custom drawing.
+ */
+public class ContactBadge extends ImageView implements OnClickListener {
```

Comment:
This class looks like it's copied from somewhere. What is the origin of this code, and is the license compatible?